satellite/console/consoleweb/server.go

```diff
@@ -192,7 +212,282 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
+	})
+}
+
+// tokenRequestHandler authenticates User by credentials and returns auth token.
+func (server *Server) tokenRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type tokenRequestModel struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+
+	var tokenRequest tokenRequestModel
+	err = json.NewDecoder(r.Body).Decode(&tokenRequest)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+	}
+
+	token, err := server.service.Token(ctx, tokenRequest.Email, tokenRequest.Password)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(token)
+	if err != nil {
+		server.log.Error(Error.Wrap(err).Error())
+	}
+}
+
+// changeAccountPasswordRequestHandler updates password for a given user.
+func (server *Server) changeAccountPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type changePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange changePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJSONError(w, 404, err)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// createNewUserRequestHandler gets password hash value and creates new inactive User.
+func (server *Server) createNewUserRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type createUserRequestModel struct {
+		console.CreateUser
+		Secret         string `json:"secret"`
+		ReferrerUserID string `json:"referrerUserId"`
+	}
+
+	var model createUserRequestModel
+	err = json.NewDecoder(r.Body).Decode(&model)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	secret, err := console.RegistrationSecretFromBase64(model.Secret)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	user, err := server.service.CreateUser(ctx, model.CreateUser, secret, model.ReferrerUserID)
+	if err != nil {
+		server.serveJSONError(w, 400, err)
+		return
+	}
+
+	token, err := server.service.GenerateActivationToken(ctx, user.ID, user.Email)
+	if err != nil {
+		server.serveJSONError(w, 500, err)
+		return
+	}
+
+	link := server.config.ExternalAddress + ActivationPath + token
+	userName := user.ShortName
+	if user.ShortName == "" {
+		userName = user.FullName
+	}
+
+	server.mailService.SendRenderedAsync(
+		ctx,
+		[]post.Address{{Address: user.Email, Name: userName}},
+		&consoleql.AccountActivationEmail{
+			Origin:         server.config.ExternalAddress,
+			ActivationLink: link,
+		},
+	)
+
+	err = json.NewEncoder(w).Encode(user)
```

Comment:
I'm wondering if we actually need to return anything. Does the client need to receive the information of the newly created user when most of them has been submitted by itself?

On the other hand, I would respond with `201` rather than `200` status code.